productions/p1.py

```diff
@@ -7,19 +7,26 @@ class P1():
     @staticmethod
     def apply(G: nx.Graph, offset = 1):
         El_node = None
-        for node in G.nodes(data=True):
-            if node[1]['label'] == 'El':
-                El_node = node
-                break
+        
+        # find isos and get pick first one
```

Comment:
All this new isomorphism logic should not be in apply.

Reply:
done m8